docs/security/inputs.md

```diff
@@ -0,0 +1,51 @@
+# Securing your inputs
+
+## `GET` parameters
+
+If you have a page or a form which reads parameters from the URL (aka `GET` parameters) like `?cid=1234` or `?action=add`, it's important to understand that attackers can somewhat easily deceive *privileged users* into submitting malicious `GET` requests by directing the user to an email or website with content like: 
+
+```html
+<img width="0" height="0" src="https://example.org/civicrm/page?foo=ATTACK" >
+```
+
+This means that we can *never* trust `GET` parameters, even if the page has tight [permissions](/security/permissions.md) or [ACLs](/security/access.md)! A common security vulnerability which arises from insecure `GET` inputs is [reflected XSS](https://excess-xss.com/#reflected-xss), but `GET` inputs can also find their way into all sort of other sensitive outputs, like SQL queries.
+
+### Validating `GET` parameters
+
+Use the function `CRM_Utils_Request::retrieve()` to retrieve and validate `GET` parameters. This works great for simple types like integers. For example:
+
+```php
+$cid = CRM_Utils_Request::retrieve('cid', 'Positive');
+```
+
+Here we have specified `'Positive'` as the type. The acceptable types can be found in [CRM_Utils_Type::validate](https://github.com/civicrm/civicrm-core/blob/60050425316acb3726305d1c34908074cde124c7/CRM/Utils/Type.php#L378).
+
+If you find yourself wanting to use the `'String'` type, beware that this type offers very little validation and hence almost no protection against attacks. Thus, for strings it's important to *add additional validation*, as demonstrated in the following example.
+
+```php
+$angPage = CRM_Utils_Request::retrieve('angPage', 'String', $this);
+if (!preg_match(':^[a-zA-Z0-9\-_/]+$:', $angPage)) {
+  CRM_Core_Error::fatal('Malformed return URL');
+}
+```
+
+## `POST` parameters
+
+When accepting `POST` parameters through forms, it's important to validate the data as 
```

Comment:
Ha! Yes. Thank you for catching that. Fixed (sort of) in e96ce38628eb7c08bbc0976eed6a81ab1de27336 although I don't actually know a whole lot about the best ways to do form validation, so I'm not sure what else to write in the section (which I guess is why I forgot to finish it!). If you have a good sense of these concepts, maybe you could add some content here? Although I think we could merge this as-is, even though it's not very complete at the moment. Then add more content later.